CDCgov/prime-reportstream
# Fixing an issue where all reports in blob storage are getting marked …

…with .internal.csv in the filename no matter what the actual format is

This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Set breakpoints on line 176 of the workflow engine, you will want to be checking the report.bodyFormat.ext - for the universal pipeline tests, the bodyFormat should also be specified correctly.
2. Run SimpleReport Test Data Copy 3 in Postman, should return .csv
3. Run SimpleReport Test Data Copy 3 in Postman, remove all but one record, should return .csv
4. Run Send CE HL7 Copy in Postman, should return .hl7
5. Remove the headers and all but one record, run Send CE HL7 Copy in Postman, should return .hl7

## Changes
- Changed the serializers to handle the bodyFormat

## Checklist

### Testing
- [Y ] Tested locally?
- [N/A ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ Y] Added tests?

### Process
- [N ] Are there licensing issues with any new dependencies introduced?
- [ Y] Includes a summary of what a code reviewer should test/verify?
- [ N] Updated the release notes?
- [ N/A] Database changes are submitted as a separate PR?
- [ N/A] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue 7935

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue N/A

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? No
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints? No
- Does this change introduce new dependencies that need vetting? No
- Does this change require changes to our infrastructure? No
- Does logging contain sensitive data? No
- Does this PR include or remove any sensitive information itself? No
